### PR TITLE
Add text correction taskflow

### PR DESCRIPTION
**Pull request by joey12300**
### PR types

New features
### PR changes

APIs
### Description

Add text correction taskflow

### CODE REVIEW

File: paddlenlp/taskflow/utils.py [hunk 1]
```diff
@@ -487,3 +487,26 @@ def save(self, save_dir):
                 node = self[nid]
                 if node.node_type == "term":
                     print(node, file=fp)
+
+
+def is_chinese_char(cp):
```
Comment (wawltor) [inline paddlenlp/taskflow/utils.py]: 可以把这个函数放到paddlenlp下面的utils去 减少重复代码

File: paddlenlp/taskflow/text_correction.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+           text_correction('遇到逆竟时，我们必须勇于面对，而且要愈挫愈勇，这样我们才能朝著成功之路前进。')
+           '''
+           [{'source': '遇到逆竟时，我们必须勇于面对，而且要愈挫愈勇，这样我们才能朝著成功之路前进。',
+             'target': '遇到逆境时，我们必须勇于面对，而且要愈挫愈勇，这样我们才能朝著成功之路前进。'}]
+           '''
+
+           text_correction('人生就是如此，经过磨练才能让自己更加拙壮，才能使自己更加乐观。')
+           '''
+           [{'source': '人生就是如此，经过磨练才能让自己更加拙壮，才能使自己更加乐观。',
+             'target': '人生就是如此，经过磨练才能让自己更加茁壮，才能使自己更加乐观。'}]
```
Comment (wawltor) [inline paddlenlp/taskflow/text_correction.py]: 可以把decode的部分再细化一下，可以出错的地方搞出一个映射表

File: paddlenlp/taskflow/text_correction.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                token_ids, token_type_ids, pinyin_ids, lengths = self.batchify_fn(
+                    examples)
+                self.input_handles[0].copy_from_cpu(token_ids)
+                self.input_handles[1].copy_from_cpu(pinyin_ids)
+                self.predictor.run()
+                det_error_probs = self.output_handle[0].copy_to_cpu()
+                corr_logits = self.output_handle[1].copy_to_cpu()
+
+                det_pred = det_error_probs.argmax(axis=-1)
+                char_preds = corr_logits.argmax(axis=-1)
```
Comment (wawltor) [inline paddlenlp/taskflow/text_correction.py]: 可以两个argmax的操作放到model里面去，减少计算开销

File: paddlenlp/taskflow/text_correction.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+from ..transformers import ErnieTokenizer, ErnieModel
+from ..transformers import is_chinese_char
+from ..datasets import load_dataset
+from ..data import Stack, Pad, Tuple, Vocab
+from .utils import download_file, add_docstrings, static_mode_guard, dygraph_mode_guard
+from .models import ErnieForCSC
+from .task import Task
+
+usage = r"""
+           from paddlenlp import Taskflow
```
Comment (wawltor) [inline paddlenlp/taskflow/text_correction.py:36]: 相应的使用文档可以放到examples下面去

Comment (joey12300) [reply paddlenlp/taskflow/text_correction.py:36]: 已放入。

File: paddlenlp/taskflow/utils.py (hunk 1 above)
Comment (joey12300) [reply paddlenlp/taskflow/utils.py]: done

File: paddlenlp/taskflow/text_correction.py (hunk 2 above)
Comment (joey12300) [reply paddlenlp/taskflow/text_correction.py]:
```
[{'source': '人生就是如此，经过磨练才能让自己更加拙壮，才能使自己更加乐观。',
    'target': '人生就是如此，经过磨练才能让自己更加茁壮，才能使自己更加乐观。',
    'errors': [{'position': 18, 'correction': {'拙': '茁'}}]}]
```
done

File: paddlenlp/taskflow/text_correction.py (hunk 3 above)
Comment (joey12300) [reply paddlenlp/taskflow/text_correction.py]: done

File: paddlenlp/transformers/tokenizer_utils.py [hunk 5]
```diff
@@ -113,36 +116,36 @@ def _is_punctuation(char):
     return False
 
 
-def tokenize_chinese_chars(text):
-    """Adds whitespace around any CJK character."""
+def is_chinese_char(cp):
```
Comment (wawltor) [inline paddlenlp/transformers/tokenizer_utils.py:119]: csc的下面是不是也可以使用用这个来替换了？

Comment (joey12300) [reply paddlenlp/transformers/tokenizer_utils.py:119]: done

File: examples/text_correction/ernie-csc/README.md [hunk 6]
```diff
@@ -133,6 +133,29 @@ Source: 人生就是如此，经过磨练才能让自己更加拙壮，才能使
 Target: 人生就是如此，经过磨练才能让自己更加茁壮，才能使自己更加乐观。
 ```
 
+### Taskflow中文一键预测
+可以使用PaddleNLP提供的Taskflow工具来对输入的文本进行一键纠错，具体使用方法如下:
+
+```python
+from paddlenlp import Taskflow
+text_correction = Taskflow("text_correction")
```
Comment (ZeyuChen) [inline examples/text_correction/ernie-csc/README.md:141]: 是否得考虑补充csc？为将来更多的应用能力做拓展留好接口

File: examples/text_correction/ernie-csc/README.md [hunk 7]
```diff
@@ -133,6 +133,29 @@ Source: 人生就是如此，经过磨练才能让自己更加拙壮，才能使
 Target: 人生就是如此，经过磨练才能让自己更加茁壮，才能使自己更加乐观。
 ```
 
+### Taskflow中文一键预测
```
Comment (ZeyuChen) [inline examples/text_correction/ernie-csc/README.md]: Taskflow一键预测
不用突出中文

File: examples/text_correction/ernie-csc/README.md (hunk 6 above)
Comment (ZeyuChen) [reply examples/text_correction/ernie-csc/README.md:141]: text-correction-csc?

File: paddlenlp/taskflow/models/text_correction_model.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import paddle.nn as nn
+
+
+class ErnieForCSC(nn.Layer):
```
Comment (ZeyuChen) [inline paddlenlp/taskflow/models/text_correction_model.py:20]: 加上注释，解释下CSC的完整含义

File: paddlenlp/taskflow/text_correction.py [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+                for i in range(len(lengths)):
+                    batch_result.append(
+                        (det_preds[i], char_preds[i], lengths[i]))
+                results.append(batch_result)
+        inputs['batch_results'] = results
+        return inputs
+
+    def _postprocess(self, inputs):
+        """
+        The model output is allways the logits and pros, this function will convert the model output to raw text.
```
Comment (ZeyuChen) [inline paddlenlp/taskflow/text_correction.py]: allways 拼错了，以及always放这里是不是很多于？不是太好理解

Comment (joey12300) [reply paddlenlp/taskflow/text_correction.py]: removed

File: examples/text_correction/ernie-csc/README.md (hunk 7 above)
Comment (joey12300) [reply examples/text_correction/ernie-csc/README.md]: done

File: paddlenlp/taskflow/models/text_correction_model.py (hunk 8 above)
Comment (joey12300) [reply paddlenlp/taskflow/models/text_correction_model.py:20]: done

**Review by ZeyuChen — APPROVED** [review]
LGTM
